lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -34,6 +34,11 @@ def as_json
 
         private
 
+        def ensure_valid!(json)
```

Comment:
```
"relationships": {
  "description": "Members of the relationships object (\"relationships\") represent references from the resource object in which it's defined to other resource objects.",
  "type": "object",
  "patternProperties": {
    "^\\w[-\\w_]*$": {
      "properties": {
        "links": {
          "$ref": "#/definitions/links"
        },
        "data": {
          "description": "Member, whose value represents \"resource linkage\".",
          "oneOf": [
            {
              "$ref": "#/definitions/relationshipToOne"
            },
            {
              "$ref": "#/definitions/relationshipToMany"
            }
          ]
        },
        "meta": {
          "$ref": "#/definitions/meta"
        }
      },
      "additionalProperties": false
    }
  },
  "additionalProperties": false
}
```

None of those are set as required. `relationships` itself isn't a required key either.

¯\_(ツ)_/¯

> we just aren't returning it in this response

This is what I'm trying to understand. The behavior you're describing seems wrong to me.

I agree that `data: null` and `data: []` convey _empty_ relationships. If you don't want to return a relationship in the response, wouldn't you just omit the entire relationship? So if you don't want to return the relationships for posts, then it seems logical that this would be conveyed by not having `posts` in `relationships` at all. And continuing onward, if there are no relationships, why have a `relationships` key? If you want to let consumers know that a relationship does exist but isn't available from the current available data, you'd probably provide a `link` for it, right? Conveying this "unknown relationships" information with empty meta seems weird to me. What if the relationship actually had meta if you were to fetch it?